osbenchmark/workload/params.py

```diff
@@ -872,6 +867,14 @@ def partition(self, partition_index, total_partitions):
         Returns:
             The parameter source for this particular partition
         """
+        if self.data_set is None:
```

Comment:
As discussed offline, this will not have impact over multithreading since partition is called by original instance and by decoupling dataset loading from constructor, get_data_set is idempotent when passed same parameters. This is also consistent with bulk para source too.